apiserver/facades/agent/meterstatus/meterstatus.go

```diff
@@ -78,72 +73,84 @@ func NewMeterStatusAPI(
 // to each unit's meter status.
+// This is a noop as of 3.6.10 because meter status functionality is removed.
 func (m *MeterStatusAPI) WatchMeterStatus(args params.Entities) (params.NotifyWatchResults, error) {
 	result := params.NotifyWatchResults{
 		Results: make([]params.NotifyWatchResult, len(args.Entities)),
 	}
-	canAccess, err := m.accessUnit()
-	if err != nil {
-		return params.NotifyWatchResults{}, err
-	}
-	for i, entity := range args.Entities {
-		tag, err := names.ParseUnitTag(entity.Tag)
-		if err != nil {
-			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
-			continue
-		}
-		err = apiservererrors.ErrPerm
-		var watcherId string
-		if canAccess(tag) {
-			watcherId, err = m.watchOneUnitMeterStatus(tag)
+	for i := range args.Entities {
+		// Create a simple notify watcher that only sends one initial event.
+		w := newEmptyNotifyWatcher()
+		watcherID := m.resources.Register(w)
+
+		result.Results[i] = params.NotifyWatchResult{
+			NotifyWatcherId: watcherID,
 		}
-		result.Results[i].NotifyWatcherId = watcherId
-		result.Results[i].Error = apiservererrors.ServerError(err)
 	}
-	return result, nil
-}
 
-func (m *MeterStatusAPI) watchOneUnitMeterStatus(tag names.UnitTag) (string, error) {
-	unit, err := m.state.Unit(tag.Id())
-	if err != nil {
-		return "", err
-	}
-	watch := unit.WatchMeterStatus()
-	if _, ok := <-watch.Changes(); ok {
-		return m.resources.Register(watch), nil
-	}
-	return "", watcher.EnsureErr(watch)
+	return result, nil
 }
 
 // GetMeterStatus returns meter status information for each unit.
+// This is a noop as of 3.6.10 because meter status functionality is removed.
 func (m *MeterStatusAPI) GetMeterStatus(args params.Entities) (params.MeterStatusResults, error) {
-	result := params.MeterStatusResults{
+	return params.MeterStatusResults{
 		Results: make([]params.MeterStatusResult, len(args.Entities)),
+	}, nil
+}
+
+// newEmptyNotifyWatcher returns starts and returns a new empty notify watcher,
+// with only the initial event.
+func newEmptyNotifyWatcher() *emptyNotifyWatcher {
+	changes := make(chan struct{})
+
+	w := &emptyNotifyWatcher{
+		changes: changes,
 	}
-	canAccess, err := m.accessUnit()
-	if err != nil {
-		return params.MeterStatusResults{}, apiservererrors.ErrPerm
-	}
-	for i, entity := range args.Entities {
-		unitTag, err := names.ParseUnitTag(entity.Tag)
-		if err != nil {
-			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
-			continue
-		}
-		err = apiservererrors.ErrPerm
-		var status state.MeterStatus
-		if canAccess(unitTag) {
-			var unit *state.Unit
-			unit, err = m.state.Unit(unitTag.Id())
-			if err == nil {
-				status, err = MeterStatusWrapper(unit.GetMeterStatus)
-			}
-			result.Results[i].Code = status.Code.String()
-			result.Results[i].Info = status.Info
+	w.tomb.Go(func() error {
+		changes <- struct{}{}
+		defer close(changes)
+		return w.loop()
+	})
```

Comment:
We need to ensure the send on this channel is non-blocking.
```suggestion
	w.tomb.Go(func() error {
		defer close(changes)
		select {
		case changes <- struct{}{}:
		case <-w.tomb.Dying():
			return tomb.ErrDying
		}
		return w.loop()
	})
```

Reply:
Addressed. Thank you.